slang-gfx.h

```diff
@@ -1333,7 +1430,14 @@ class IComputeCommandEncoder : public ICommandEncoder
         SLANG_RETURN_NULL_ON_FAIL(bindPipeline(state, &rootObject));
         return rootObject;
     }
+
+    // Sets the current pipeline state and bind an existing root shader object.
+    virtual SLANG_NO_THROW Result SLANG_MCALL bindPipelineAndRootObject(
+        IPipelineState* state,
+        IShaderObject* rootObject) = 0;
```

Comment:
For the compute case specifically, it seems redundant to have to specify both the pipeline state *and* a root shader object. The root shader object would have been created from a specific `IShaderProgram`, and pretty much the only things the `IPipelineState` stores for the compute case is the `IShaderProgram`.

I guess this is really just me getting to the point that maybe we don't need `IPipelineState`s for compute or, if we do, there could be some streamlined/simplified options where people can dispatch compute just using the `IShaderProgram` (or a pre-existing root object) more directly.

Reply:
This will need to be done in a separate pass to get rid of all compute pipeline states.